tests/light/src/syslog_ng_config/statements/destinations/unix_dgram_destination.py

```diff
@@ -33,9 +32,9 @@
 
 
 class UnixDgramDestination(DestinationDriver):
-    def __init__(self, file_name, **options):
+    def __init__(self, testcase_parameters, file_name, **options):
         self.driver_name = "unix-dgram"
-        self.path = Path(tc_parameters.WORKING_DIR, file_name)
+        self.path = Path(testcase_parameters.get_relative_working_dir(), file_name)
```

Comment:
I think it is a user-side (GitHub Actions) problem, and should be fixed on the user side.

Can't we just move the reports dir to somewhere else, with a shorter path in GitHub Actions and/or make the socker name shorter in the TCs (which at least does not change the test framework, only the test cases)?

```py
def calculate_working_dir(pytest_config_object, testcase_name):
    report_dir = Path(pytest_config_object.getoption("--reports")).resolve().absolute()
    return Path(report_dir, calculate_testcase_name(testcase_name))

...

def pytest_runtest_setup(item):
    logging_plugin = item.config.pluginmanager.get_plugin("logging-plugin")
    tc_parameters.WORKING_DIR = working_dir = calculate_working_dir(item.config, item.name)
    ...
```

Reply:
Yes of-course we can customize the reports dir path with a command line argument: '--reports' (we already have this). But I think this is a more specific case. From users point of view in one set of testcases Light can started with no error, but in an another set it could not (although everything is correct from testcase side). However Light is responsible for generating file-paths for drivers (so Light has created this long path), and because of this I think Light should handle this long path issues. I like to give a general fix for this as user should not change the execution arguments for Light. One more thing related to this we already fix this for ctl control_socket_path in a same way at: https://github.com/syslog-ng/syslog-ng/blob/c97bdf2ea26f446560c7515c6ab42283eafac140/tests/light/src/syslog_ng/syslog_ng_paths.py#L51

Reply:
@alltilla  Thanks for the discussion and the suggestions. 
We have discussed this irl. In this iteration testcase should handle the correct path length for sockets (not the internal config driver classes).
We also concluded in a new concept, where we hope we can remove the tc_parameters.WORKING_DIR global variable.
This will be fixed in a separate PR.

Reply:
We concluded, that most of (if not all) `WORKING_DIR` related path calculating logic can be eliminated, if we changed the CWD to the testcase's working dir at the start of each testcase.